File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/inline/uncomment/direct/Java/Go/s012761786_commented.go
package main

import (
	"fmt"
	"math"
)

func main() {
	var n, a, b int64
	fmt.Scanf("%d %d %d", &n, &a, &b)
	sa := int64(math.Abs(float64(a - b)))

	var ans int64 = 0

	if sa%2 == 0 {
		ans = sa / 2
	} else {
		AAA := int64(math.Min(float64(n-a), float64(a-1)))
		BBB := int64(math.Min(float64(n-b), float64(b-1)))

		ans = int64(math.Min(float64(AAA), float64(BBB))) // 片方が端っこに来るまで。

		// 差を1減らす⇒差が偶数になる
		sa -= 1
		ans += 1

		// 差は偶数
		ans += sa / 2
	}

	fmt.Println(ans)
}

// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/inline/comment/direct/Java/C/s012761786_commented.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    long n, a, b, sa, ans;
    scanf("%ld", &n);
    scanf("%ld", &a);
    scanf("%ld", &b);

    sa = labs(a - b);

    ans = 0;

    if (sa % 2 == 0) {
        ans = sa / 2;
    } else {
        long AAA = (n - a) < (a - 1) ? (n - a) : (a - 1);
        long BBB = (n - b) < (b - 1) ? (n - b) : (b - 1);
        ans = (AAA < BBB) ? AAA : BBB;

        sa -= 1;
        ans += 1;
        ans += sa / 2;
    }

    printf("%ld\n", ans);

    return 0;
}
// End of Code
